### PR TITLE
V22F-304 Linting

### PR DESCRIPTION
**Pull request by KungFuDonkey**

### CODE REVIEW

**Comment by KungFuDonkey** [comment]
image

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Install linter package
+        run: python3 -m pip install git+https://github.com/shenxianpeng/cpp-linter-action
+
+      - uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x86
+
+      - name: Generate compilation database  
+        run: cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DOPTION_PACKAGING=0 -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -S '${{github.workspace}}/speed-dreams/source-2.2.3' -B '${{runner.workspace}}/build' -G Ninja
```
Comment (suppergerrie2) [inline .github/workflows/linting.yml:30]:
```suggestion
        run: cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DOPTION_PACKAGING=0 -S '${{github.workspace}}/speed-dreams/source-2.2.3' -B '${{runner.workspace}}/build' -G Ninja
```

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Fail fast?!
+        if: steps.linter.outputs.checks-failed > 0
+        run: |
+          echo "Some files failed the linting checks!"
+        # for actual deployment
+        # run: exit 1
```
Comment (suppergerrie2) [inline .github/workflows/linting.yml:46]:
```suggestion
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/LinterTests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/LinterTests/CMakeLists.txt:1]: Is this folder/file still needed?

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by suppergerrie2 — APPROVED** [review]

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**